src/BizHawk.Emulation.Cores/Consoles/3DO/Opera.cs

```diff
@@ -0,0 +1,341 @@
+			}, new Delegate[] { _CDReadCallback, _CDSectorCountCallback });
+
+			// Setting CD callbacks
+			_libOpera.SetCdCallbacks(_CDReadCallback, _CDSectorCountCallback);
+
+			// Adding BIOS file
+			string biosType = _syncSettings.SystemType switch
+			{
+				SystemType.Panasonic_FZ1_U => "Panasonic_FZ1_U",
+				SystemType.Panasonic_FZ1_E => "Panasonic_FZ1_E",
+				SystemType.Panasonic_FZ1_J => "Panasonic_FZ1_J",
+				SystemType.Panasonic_FZ10_U => "Panasonic_FZ10_U",
+				SystemType.Panasonic_FZ10_E => "Panasonic_FZ10_E",
+				SystemType.Panasonic_FZ10_J => "Panasonic_FZ10_J",
+				SystemType.Goldstar_GDO101P => "Goldstar_GDO101P",
+				SystemType.Goldstar_FC1 => "Goldstar_FC1",
+				SystemType.Sanyo_IMP21J_Try => "Sanyo_IMP21J_Try",
+				SystemType.Sanyo_HC21 => "Sanyo_HC21",
+				SystemType.Shootout_At_Old_Tucson => "Shootout_At_Old_Tucson",
+				SystemType._3DO_NTSC_1fc2 => "3DO_NTSC_1fc2",
+				_ => "None"
+			};
+
+			var (biosData, biosInfo) = CoreComm.CoreFileProvider.GetFirmwareWithGameInfoOrThrow(new(VSystemID.Raw._3DO, biosType), "BIOS ROM files are required!");
+			string biosFileName = biosInfo.Name + ".bin";
+			_exe.AddReadonlyFile(biosData, biosFileName);
+
+			// Adding Font ROM file, if required
+			string fontROMType = _syncSettings.FontROM switch
+			{
+				FontROM.Kanji_ROM_Panasonic_FZ1 => "Kanji_ROM_Panasonic_FZ1",
+				FontROM.Kanji_ROM_Panasonic_FZ10 => "Kanji_ROM_Panasonic_FZ10",
+				_ => "None"
+			};
+
+			string fontROMFileName = "None";
+			if (fontROMType != "None")
+			{
+				var (fontROMData, fontROMInfo) = CoreComm.CoreFileProvider.GetFirmwareWithGameInfoOrThrow(new(VSystemID.Raw._3DO, fontROMType), "Font ROM files are required!");
+				_exe.AddReadonlyFile(fontROMData, fontROMInfo.Name);
+				fontROMFileName = fontROMInfo.Name;
+			}
+
+			////////////// Initializing Core
+			string cdName = _discAssets[0].DiscName;
+			Console.WriteLine($"Launching Core with Game: '{cdName}', BIOS ROM: '{biosFileName}', Font ROM: '{fontROMFileName}'");
+			if (!_libOpera.Init(cdName, biosFileName, fontROMFileName, (int)_syncSettings.Controller1Type, (int)_syncSettings.Controller2Type, (int)_syncSettings.VideoStandard))
+				throw new InvalidOperationException("Core rejected the rom!");
+
+			PostInit();
+		}
+
+		// CD Handling logic
+		private bool _isMultidisc;
+		private bool _discInserted = true;
+		private readonly LibOpera.CDReadCallback _CDReadCallback;
+		private readonly LibOpera.CDSectorCountCallback _CDSectorCountCallback;
+		private int _discIndex;
+		private readonly List<DiscSectorReader> _cdReaders = new List<DiscSectorReader>();
+		private static int CD_SECTOR_SIZE = 2048;
+		private readonly byte[] _sectorBuffer = new byte[CD_SECTOR_SIZE];
+
+		private void SelectNextDisc()
+		{
+			_discIndex++;
+			if (_discIndex == _discAssets.Count) _discIndex = 0;
+			CoreComm.Notify($"Selected CDROM {_discIndex}: {_discAssets[_discIndex].DiscName}", _messageDuration);

```

Comment:
The intent of the duration param is for a human to step in and keep long strings on-screen longer. If you're not customising them individually, pass `null`, not `4`.

Reply:
fixed